src/App.js

```diff
@@ -106,34 +107,36 @@ const App = compose(injectState, withApi)(({ editing, setEditing, state, api })
             })
           }
         />
-        <Route
-          path="/join"
-          exact
-          render={props => {
-            return (
+        <ApiContext.Provider value={initializeApi()}>
```

Comment:
ah yeah, actually this is a bit naive, i'll touch it up. basically we need different logic on these pages for handling errors. login page handles its own, displaying an 'unauthorized message', so it doesn't need a special global handler.

join, on the other hand, should probably just redirect back to login on 401 or 403, i'll make that change

Reply:
the most important thing here was that these are not wrapped in force select role, so having the window.location.reload() handler was causing an infinite reload loop lol

Reply:
yeah `window.location.reload()` was a naive catch-all placeholder to get it to work, but probably would make more sense to display something before kicking users off :P

Reply:
yeah for sure, i know it's to be changed anyway, but I think those pages do kind of have inherently different 'unauthorized' logic than the others, since we haven't actually vetted that the user has access yet